src/Worker/DBUpdate.php

```diff
@@ -6,17 +6,12 @@
 namespace Friendica\Worker;
 
 use Friendica\Core\Config;
+use Friendica\Core\Update;
 
 class DBUpdate
 {
 	public static function execute()
 	{
-		$a = \Friendica\BaseObject::getApp();
-
-		// We are deleting the latest dbupdate entry.
-		// This is done to avoid endless loops because the update was interupted.
-		Config::delete('database', 'dbupdate_'.DB_UPDATE_VERSION);
```

Comment:
Why have you removed this functionality?

Reply:
I believe this is because he followed my suggestion to have a single config value holding the current DB update version of the Friendica node. Since we only have a one-way forward update mechanism, having multiple config keys probably didn't make sense.

Reply:
yep, as @MrPetovan said we now have 4 config entries:
- `last_successful_update = DB_UPDATE_VERSION`
- `last_successful_update_time = time()`
- `last_successful_update_function = $funcname`
- `last_successful_update_function_time = time()`

Reply:
So this means that we not only have to test the whole update mechanism, but also we have to test the condition that lead to this code that had been removed.

Thing is: We have to avoid that the update mechanism can be called when the updater mechanism is already running. This is crucial, since it is called automatically at page load.

But: When something is going wrong, we have to be able to override it by hand.

We had some issue here lately that was (most likely) caused by calling the structure update by command line while it already had been started from index.php. So I suggest that we check if it is running and only if you add some "--override" (or similar) you can start it again.